overlord/snapstate/handlers.go

```diff
@@ -2793,6 +2793,10 @@ func (m *SnapManager) doClearSnapData(t *state.Task, _ *tomb.Tomb) error {
 			return err
 		}
 
+		if err := m.backend.RemoveUserSnapDir(snapsup.InstanceName(), dirOpts); err != nil {
```

Comment:
Maybe add a comment above this line saying something like:
```
At this point all snap data under snap's user directory got removed by RemoveSnapData, remove the directory itself```

Reply:
@MiguelPires what has changed with respect to the old world that we need this now? before we weren't doing this?